Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-327.28.3.el7.x86_64)
 GCC:gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
Sha: 64d8c38c3d9ceabae24709a6dfc2679339a47586
Number of Error(s): 7
/mnt/disk1/home/vamosax/smoketest/PR-12916/HPCC-Platform/esp/services/ws_dfu/ws_dfuService.cpp:374:102: error: \xe2\x80\x98struct IEspDFUInfoRequest\xe2\x80\x99 has no member named \xe2\x80\x98getRestricted\xe2\x80\x99; did you mean \xe2\x80\x98getRestrict\xe2\x80\x99?
/mnt/disk1/home/vamosax/smoketest/PR-12916/HPCC-Platform/esp/services/ws_dfu/ws_dfuService.cpp:377:102: error: \xe2\x80\x98struct IEspDFUInfoRequest\xe2\x80\x99 has no member named \xe2\x80\x98getRestricted\xe2\x80\x99; did you mean \xe2\x80\x98getRestrict\xe2\x80\x99?
/mnt/disk1/home/vamosax/smoketest/PR-12916/HPCC-Platform/esp/services/ws_dfu/ws_dfuService.cpp:382:102: error: \xe2\x80\x98struct IEspDFUInfoRequest\xe2\x80\x99 has no member named \xe2\x80\x98getRestricted\xe2\x80\x99; did you mean \xe2\x80\x98getRestrict\xe2\x80\x99?
/mnt/disk1/home/vamosax/smoketest/PR-12916/HPCC-Platform/esp/services/ws_dfu/ws_dfuService.cpp:1994:51: error: \xe2\x80\x98CDFUChangeRestricted_RequireSigned\xe2\x80\x99 was not declared in this scope
make[2]: *** [esp/services/ws_dfu/CMakeFiles/ws_dfu.dir/build.make:128: esp/services/ws_dfu/CMakeFiles/ws_dfu.dir/ws_dfuService.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:11143: esp/services/ws_dfu/CMakeFiles/ws_dfu.dir/all] Error 2
make: *** [Makefile:163: all] Error 2

[Build error:](https://gist.github.com/23a4ee16e382aec73da8c974d1dd93c0)
Build log [Build log](https://gist.github.com/HPCCSmoketest/23a4ee16e382aec73da8c974d1dd93c0#file-result-19-08-30-05-14-42-log)

--- a/esp/scm/ws_dfu.ecm
+++ b/esp/scm/ws_dfu.ecm
@@ -32,11 +32,11 @@ ESPenum DFUChangeProtection : int
 };
 
 
-ESPenum DFUChangeRestricted : int
+ESPenum DFUChangeRestriction : int
 {
     NoChange(0, "No Change"),
-    RequireSigned(1, "Restricted"),
-    NotRestricted(2, "NotRestricted")
+    Restrict(1, "Restrict"),
+    NotRestricted(2, "Unrestricted")
 };
 
 ESPenum DFUDefFileFormat : string
@@ -372,7 +372,7 @@ DFUInfoRequest
     [min_ver("1.37")] bool IncludeJsonTypeInfo(false);
     [min_ver("1.37")] bool IncludeBinTypeInfo(false);
     [min_ver("1.39")] ESPenum DFUChangeProtection Protect(0);
-    [min_ver("1.53")] ESPenum DFUChangeRestricted Restricted(0);
+    [min_ver("1.53")] ESPenum DFUChangeRestriction Restrict(0);
 };
 
 ESPresponse

--- a/esp/services/ws_dfu/ws_dfuService.cpp
+++ b/esp/services/ws_dfu/ws_dfuService.cpp
@@ -371,15 +371,15 @@ bool CWsDfuEx::onDFUInfo(IEspContext &context, IEspDFUInfoRequest &req, IEspDFUI
             double version = context.getClientVersion();
             if (version < 1.38)
                 doGetFileDetails(context, userdesc.get(), req.getFileName(), req.getCluster(), req.getQuerySet(), req.getQuery(), req.getFileDesc(),
-                    req.getIncludeJsonTypeInfo(), req.getIncludeBinTypeInfo(), req.getProtect(), req.getRestricted(), resp.updateFileDetail());
+                    req.getIncludeJsonTypeInfo(), req.getIncludeBinTypeInfo(), req.getProtect(), req.getRestrict(), resp.updateFileDetail());
             else
                 doGetFileDetails(context, userdesc.get(), req.getName(), req.getCluster(), req.getQuerySet(), req.getQuery(), req.getFileDesc(),
-                    req.getIncludeJsonTypeInfo(), req.getIncludeBinTypeInfo(), req.getProtect(), req.getRestricted(), resp.updateFileDetail());
+                    req.getIncludeJsonTypeInfo(), req.getIncludeBinTypeInfo(), req.getProtect(), req.getRestrict(), resp.updateFileDetail());
         }
         else
         {
             doGetFileDetails(context, userdesc.get(), req.getName(), req.getCluster(), req.getQuerySet(), req.getQuery(), NULL,
-                    req.getIncludeJsonTypeInfo(), req.getIncludeBinTypeInfo(), req.getProtect(), req.getRestricted(), resp.updateFileDetail());
+                    req.getIncludeJsonTypeInfo(), req.getIncludeBinTypeInfo(), req.getProtect(), req.getRestrict(), resp.updateFileDetail());
         }
     }
     catch(IException* e)
@@ -1958,7 +1958,7 @@ void CWsDfuEx::queryFieldNames(IEspContext &context, const char *fileName, const
 
 void CWsDfuEx::doGetFileDetails(IEspContext &context, IUserDescriptor *udesc, const char *name, const char *cluster,
     const char *querySet, const char *query, const char *description, bool includeJsonTypeInfo, bool includeBinTypeInfo,
-    CDFUChangeProtection protect, CDFUChangeRestricted changeRestricted, IEspDFUFileDetail &FileDetails)
+    CDFUChangeProtection protect, CDFUChangeRestriction changeRestriction, IEspDFUFileDetail &FileDetails)
 {
     if (!name || !*name)
         throw MakeStringException(ECLWATCH_MISSING_PARAMS, "File name required");
@@ -1988,10 +1988,10 @@ void CWsDfuEx::doGetFileDetails(IEspContext &context, IUserDescriptor *udesc, co
             protectBy.set("hpcc");
         df->setProtect(protectBy.str(), protect == CDFUChangeProtection_Protect ? true : false);
     }
-    if (changeRestricted != CDFUChangeRestricted_NoChange)
+    if (changeRestriction != CDFUChangeRestriction_NoChange)
     {
         context.ensureFeatureAccess("DFURestrictedAccess", SecAccess_Write, ECLWATCH_DFU_ACCESS_DENIED, "DFURestrictedAccess: Permission denied.");
-        df->setRestrictedAccess(changeRestricted==CDFUChangeRestricted_RequireSigned);
+        df->setRestrictedAccess(changeRestriction==CDFUChangeRestriction_Restrict);
     }
 
     offset_t size=queryDistributedFileSystem().getSize(df), recordSize=df->queryAttributes().getPropInt64("@recordSize",0);

--- a/esp/services/ws_dfu/ws_dfuService.hpp
+++ b/esp/services/ws_dfu/ws_dfuService.hpp
@@ -197,7 +197,7 @@ private:
     bool doLogicalFileSearch(IEspContext &context, IUserDescriptor* udesc, IEspDFUQueryRequest & req, IEspDFUQueryResponse & resp);
     void doGetFileDetails(IEspContext &context, IUserDescriptor* udesc, const char *name,const char *cluster,
         const char *querySet, const char *query, const char *description, bool includeJsonTypeInfo, bool includeBinTypeInfo,
-        CDFUChangeProtection protect, CDFUChangeRestricted changeRestricted, IEspDFUFileDetail& FileDetails);
+        CDFUChangeProtection protect, CDFUChangeRestriction changeRestriction, IEspDFUFileDetail& FileDetails);
     bool createSpaceItemsByDate(IArrayOf<IEspSpaceItem>& SpaceItems, const char * interval, unsigned& yearFrom,
         unsigned& monthFrom, unsigned& dayFrom, unsigned& yearTo, unsigned& monthTo, unsigned& dayTo);
     bool setSpaceItemByScope(IArrayOf<IEspSpaceItem>& SpaceItems64, const char*scopeName, const char*logicalName, __int64 size);

